src/inmanta/data/__init__.py

```diff
@@ -5444,6 +5444,7 @@ class EnvironmentMetricsGauge(BaseDocument):
     :param environment: the environment to which this metric is related
     :param metric_name: The name of the metric
     :param timestamp: The timestamps at which a new record is created
+    :grouped_by: the field on which the metrics is grouped by
```

Comment:
I don't find this message all that clear.

Reply:
how would you explain it? The name is self explanatory to me :p 
Should I add an example, like:
For example the 'agent_count' metric is grouped_by "status".

Reply:
Well, the concept of grouping is just a method to do the aggregation in SQL, so in the context of the API, it took me a while to see what it referred to, because it has little meaning there, does it? For example, a field called `status` would be a lot more meaningful in my opinion, because that is exactly what it is, while the grouping is just a means to an end.

Reply:
It is of course possible I'm missing something. I haven't seen all the code for the metrics.

Reply:
We could be grouping on something else than "status" for another metric. This fields indicates which fields (in addition to the environment) was used in the sql query to aggregate the data. (Maybe this is how I should explain it :p )

Reply:
Yes, I get that, but I did overlook something else that is now clear to me. I still find it very strange but I'm having difficulty explaining it. I'll get back to this next week.

Reply:
So, with some time to let this sink in, I feel less strongly about it. The main thing that I still find unintuitive about the name (that we can clear up in the docstring) is that semantically, afaik, you generally group by "some generic name/type", which gives you categories with specific values for the thing. e.g. suppose you have something with colors, you say you group by color, giving you categories red, green, blue. You don't say you group by red.

This is why in the context of a single metric value, I find the name non trivial: in natural language it doesn't make sense in this context, so if you don't know up front what it means, it might take a while to make the connection. Therefore, I think a natural language docstring could help clarify this. e.g. "The name of the group/category this metric represents (e.g. red if grouped by color). None iff metrics of this type are not divided in groups."

I realize I wasn't making much sense before. I hope this is more clear. What do you think?

Reply:
yes! now I see what you mean and how it could be confusing. I'll change the docstring

Reply:
I have been following this from a distance, what about "group" or "category" instead of "grouped_by" ?
Also what about multiple aggregation ? Afaik we are supporting it so it might be worth adding to the docstring ?

Reply:
> what about "group" or "category" instead of "grouped_by"

I would honestly prefer that. The only reason I didn't propose it myself is because I didn't want to be too heavy on suggesting changes to existing code. But if it's feasible and everyone agrees I'm definitely in favor.

>  Also what about multiple aggregation ?

What exactly do you mean by this? To have a second axis to group by? I'm not sure we do support that at the moment.

Reply:
should be feasible I think, and I have no problem changing it :)

Reply:
> To have a second axis to group by?
Yes, I might be completely off here, but when defining a `get_metric_value` method, what prevents us from returning some  MetricValue(grouped_by="a,b") for example ?

Reply:
Nothing prevents you, but do we want to endorse that? Either we don't officially support multiple aggregation for now, or we need to do it right. In the second case it's probably better to go for a `tuple[str]` than `str`, but since we don't need it at the moment (and I don't immediately see a future use case), I feel like it would be an overengineered solution (even if it would make the migration simpler if ever we do want to add support). So while I agree with you that it could be done, I'm not sure we should.

Reply:
(cc @arnaudsjs, from Hugo's first comment (5 up) on, the rest has been settled)